opm/autodiff/GeoProps.hpp

```diff
@@ -142,6 +142,12 @@ namespace Opm
             std::vector<double> mult;
             multiplyHalfIntersections_(grid, eclState, ntg, htrans, mult);
 
+            // Handle NNCs
+            std::shared_ptr<const NNC> nnc_deck = eclState ? eclState->getNNC()
+                             : std::shared_ptr<const Opm::NNC>();
+            NNC nnc (*nnc_deck);
+            nnc_ = nnc;
```

Comment:
The code above is a bit confusing. If there is no eclState, then the nnc_ member is already (correctly) default-initialised. Otherwise, we should copy from the eclState. Something like:
```
if (eclState) {
    nnc_ = *(eclState->getNNC()); // Parenthesis not strictly needed, just to show structure.
}
```